java/code/src/com/redhat/rhn/common/db/datasource/xml/ChannelReport_queries.xml

```diff
@@ -0,0 +1,98 @@
+<!--
+  ~ Copyright (c) 2022 SUSE LLC
+  ~
+  ~ This software is licensed to you under the GNU General Public License,
+  ~ version 2 (GPLv2). There is NO WARRANTY for this software, express or
+  ~ implied, including the implied warranties of MERCHANTABILITY or FITNESS
+  ~ FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+  ~ along with this software; if not, see
+  ~ http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+  ~
+  ~ Red Hat trademarks are not licensed under GPLv2. No permission is
+  ~ granted to use or replicate Red Hat trademarks that are incorporated
+  ~ in this software or its documentation.
+  -->
+<datasource_modes>
+
+
+<mode name="Channel" class="">
+    <query>
+          SELECT rhnchannel.id AS channel_id
+                      , rhnchannel.name
+                      , rhnchannel.label
+                      , CASE 
+                          WHEN rhnchannel.org_id IS NULL THEN 'Vendor'
+                          ELSE 'Custom'
+                        END AS type
+                      , rhnproductname.name AS product_name
+                      , rhnchannelarch.name AS arch
+                      , rhnchannel.description
+                      , rhnparentchannel.label AS parent_channel_label
+                      , web_customer.name AS organization 
+            FROM rhnchannel
+                      LEFT JOIN rhnproductname ON rhnchannel.product_name_id = rhnproductname.id
+                      LEFT JOIN rhnchannelarch ON rhnchannel.channel_arch_id = rhnchannelarch.id
+                      LEFT JOIN rhnchannel rhnparentchannel ON rhnchannel.parent_channel = rhnparentchannel.id
+                      LEFT JOIN web_customer ON rhnchannel.org_id = web_customer.id
+  </query>
+</mode>
+
+<mode name="Errata" class="">
+    <query>
+          SELECT rhnerrata.id AS errata_id
+                      , rhnerrata.advisory_name
+                      , rhnerrata.advisory_type
+                      , rhnerrata.issue_date
+                      , rhnerrata.update_date
+                      , rhnerrata.product
+                      , rhncve.name AS cve
+                      , rhnerrata.synopsis
+                      , rhnchannel.label AS channel_label
+                      , web_customer.name AS organization 
```

Comment:
We could add also severity and we may want to add some of our special status codes:

- requires reboot (reboot_suggested keyword)
- affects package manager (restart_suggested keyword)
- advisory_status (retracted patches)

Reply:
I added all of them as Y/N flags.